sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StreamingWriteFn.java

```diff
@@ -125,6 +148,20 @@ public void finishBundle(FinishBundleContext context) throws Exception {
     for (ValueInSingleWindow<ErrorT> row : failedInserts) {
       context.output(failedOutputTag, row.getValue(), row.getTimestamp(), row.getWindow());
     }
+
+    if (histogram.getTotalCount() > options.getLatencyLoggingFrequency()) {
```

Comment:
Any reason to suspect that this will not log, if we fail a request snd don't attempt further requests? A logger on a periodic interval might make more sense?

Reply:
Changed to use seconds instead of the number of requests. I think setting frequency in time unit is easier and more intuitive.